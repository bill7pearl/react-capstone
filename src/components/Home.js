import React, { useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import { fetchGames } from '../redux/reducers/gamesReducer';

const Home = () => {
    const games = useSelector((state) => state.games.game);
    const [gameInput, setGameInput] = useState('');
    const dispatch = useDispatch();

    const searchGame = (e) => {
        setGameInput(e.target.value);
      };

    useEffect(() => {
      dispatch(fetchGames());
    }, [dispatch]);

      let filteredGames = games ? games.slice(0, 25): [];
     if (gameInput.trim().length > 0) {
      filteredGames = games.filter((obj) => (
        obj.title.toLowerCase().includes(gameInput.toLowerCase())
      ));
    } 
    return (
      <main className="games">
        <input type="text"  value={gameInput} placeholder="Search your game by name.." onChange={searchGame}/>
            {filteredGames.map((game) => (
            <div className="single-game" key={game.id}>
              <Link to={`/Details/${game.title}`}>
                <img className="img-api" src={game.image} alt={game.title} />
                <p className="game-title">{game.title}</p>
              </Link>
            </div>
          ))} 
      </main>
    );
        };

export default Home;