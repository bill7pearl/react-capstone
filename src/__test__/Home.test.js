import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Home from '../components/Home';
import store from '../redux/configureStore';

it('matches snapshot', () => {
  const tree = renderer.create(
    <Provider store={store}><Home /></Provider>,
  )
    .toJSON();
  expect(tree).toMatchSnapshot();
});