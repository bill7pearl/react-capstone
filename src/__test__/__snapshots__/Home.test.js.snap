// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<main
  className="games"
>
  <input
    onChange={[Function]}
    placeholder="Search your game by name.."
    type="text"
    value=""
  />
</main>
`;
